### PR TITLE
Add pomelo support (@username)

### PR DESCRIPTION
**Pull request by VelvetToroyashi**
https://github.com/discord/discord-api-docs/pull/6130

### CODE REVIEW

**Comment by Nihlus** [comment]
Ah, sorry, did not remember this PR was open. I've implemented the field itself - could you rebase your changes and change this PR into a fix for the CDN methods only?

**Comment by VelvetToroyashi** [comment]
Can do.

File: Backend/Remora.Discord.API/API/CDN.cs [hunk 1]
```diff
@@ -493,19 +494,20 @@ public static Result<Uri> GetDefaultUserAvatarUrl
...
-    /// <param name="userDiscriminator">The user's discriminator.</param>
+    /// <param name="userDiscriminatorOrID">The user's discriminator, or their ID.</param>
     /// <param name="imageFormat">The requested image format.</param>
     /// <param name="imageSize">The requested image size. May be any power of two between 16 and 4096.</param>
     /// <returns>A result which may or may not have succeeded.</returns>
+    /// <remarks>The ID of the user should be preferred, as any migrated accounts will have a discriminator of 0, and the result will be wrong.</remarks>
     public static Result<Uri> GetDefaultUserAvatarUrl
     (
-        ushort userDiscriminator,
+        OneOf<Snowflake, ushort> userDiscriminatorOrID,
```
Comment (Nihlus) [inline Backend/Remora.Discord.API/API/CDN.cs:510]: Should we still support discriminators here? Won't using the user's ID work in all cases?

**Review by Nihlus — CHANGES_REQUESTED** [review]
You've got some compile errors to fix.

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs [hunk 2]
```diff
@@ -20,6 +20,7 @@
 //  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 //
 
+using System;
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs:23]: Added a using without other code changes?

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs [hunk 3]
```diff
@@ -39,7 +39,7 @@ public interface IUser : IPartialUser
     new Snowflake ID { get; }
 
     /// <summary>
-    /// Gets the username of the user. This is not a unique value.
+    /// Gets the username of the user. This is a unique value.
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs:42]: It's not a unique value *yet* - we should avoid this wording until the migration is complete.

File: Tests/Remora.Discord.Gateway.Tests/Constants.cs [hunk 4]
```diff
@@ -37,7 +37,7 @@ public static class Constants
     public static IUser BotUser { get; } = new User
     (
         DiscordSnowflake.New(0),
-        "mock-bot",
```
Comment (Nihlus) [inline Tests/Remora.Discord.Gateway.Tests/Constants.cs:40]: Any particular reason for this change?

Comment (VelvetToroyashi) [reply Tests/Remora.Discord.Gateway.Tests/Constants.cs:40]: Compiler error, though it seems me trying to reflect upstream hasn't gone well

Comment (VelvetToroyashi) [reply Tests/Remora.Discord.Gateway.Tests/Constants.cs:40]: However it's worth noting that hyphens aren't ~technically~ supported in pomelo

File: Backend/Remora.Discord.API/API/CDN.cs (hunk 1 above)
Comment (VelvetToroyashi) [reply Backend/Remora.Discord.API/API/CDN.cs:510]: Discussed on Discord; bots still retain discrims for the foreseeable future

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs (hunk 2 above)
Comment (VelvetToroyashi) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs:23]: `[Obsolete]` attribute

Comment (Nihlus) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IPartialUser.cs:23]: Yeah but nothing else has changed in the file, and `main` builds

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs (hunk 3 above)
Comment (VelvetToroyashi) [reply Backend/Remora.Discord.API.Abstractions/API/Objects/Users/IUser.cs]: Git appears to have eaten my code
image

**Pull request merged**